nixos/modules/config/theming.nix

```diff
@@ -0,0 +1,379 @@
+
+      in {
+        # Standard theme engines for Gtk2
+        environment.systemPackages = [ pkgs.gtk2 ]
+            ++ optionalPackage gcfg.font
+            ++ optionalPackage gcfg.theme
+            ++ optionalPackage gcfg.iconTheme;
+
+        environment.etc."xdg/gtk-2.0/gtkrc".text =
+          concatStringsSep "\n" (
+            mapAttrsToList toGtk2Ini settings
+          );
+
+        environment.etc."xdg/gtk-3.0/settings.ini".text =
+          toGtk3Ini { Settings = settings; };
+
+        # disabled until https://github.com/NixOS/nixpkgs/issues/54150 is fixed
+        # in case of XSettings daemon or Wayland environment
+        programs.dconf.enable = mkDefault true;
+        environment.etc."dconf/db/site.d/00_gtk_theme".text =
+          toDconfIni { "org/gnome/desktop/interface" = keyfile; };
+        #environment.etc."dconf/profile/user".text = ''
+        #  user-db:user
+        #  system-db:site
+        #'';
+        systemd.services.dconf-update = {
+          description = "Updates system-db after site settings are changed";
+          #wantedBy = [ "multi-user.target" ];
+          restartTriggers = [ config.environment.etc."dconf/db/site.d/00_gtk_theme".source ];
+          serviceConfig = {
+            Type = "oneshot";
+            RemainAfterExit = true;
+            ExecStart = "${pkgs.gnome3.dconf}/bin/dconf update";
+          };
+        };
+      })
+    )
+
+    (mkIf qcfg.enable
+      (let
+        general =
+          optionalAttrs (qcfg.font != null)
+            {
+              font = qcfg.font.name;
+              menuFont = qcfg.font.name;
+              toolBarFont = qcfg.font.name;
+            }
+          //
+          optionalAttrs (qcfg.scheme != null)
+            { ColorScheme = qcfg.scheme.name; };
+        kde =
+          optionalAttrs (qcfg.theme != null)
+            { widgetStyle = qcfg.theme.name; };
+        icons =
+          optionalAttrs (qcfg.iconTheme != null)
+            { Theme = qcfg.iconTheme.name; };
+
+        qt =
+          optionalAttrs (qcfg.font != null)
+            { font = ''"${qcfg.font.name}"''; }
+          //
+          optionalAttrs (gcfg.theme != null)
+            { style = "GTK+"; };
+      in {
+        # because QKdeTheme doesn't read ColorScheme key
+        environment.systemPackages = [ pkgs.plasma-integration ]
+            ++ optionalPackage qcfg.font
+            ++ optionalPackage qcfg.theme
+            ++ optionalPackage qcfg.iconTheme;
+
+        environment.pathsToLink = [
+          "/share/color-schemes"
+        ];
+
+        # Qt4
+        environment.etc."xdg/Trolltech.conf".text =
+          toGtk3Ini {
+            Qt = qt;
+          };
+
+        # Qt5
+        environment.etc."xdg/kdeglobals".text =
+          toGtk3Ini {
+            General = general;
+            KDE = kde;
+            Icons = icons;
+          };
+
+        environment.variables.QT_QPA_PLATFORMTHEME = "kde";
+      })
+    )
+
+    (mkIf (theme == "adwaita") {
```

Comment:
@worldofpeace 
If I split this module, where should I put these predefined theme profiles that create unified look?

Reply:
I'd prefer if we could keep the PR smaller and not have any defaults like that yet.
Things like this aren't managed declaratively yet, so it might make the feature more difficult to land.